torch/_decomp/decompositions.py

```diff
@@ -1902,13 +1902,21 @@ def upsample_bilinear2d_vec(
     y_ceil = torch.ceil(y).clamp(max=in_w - 1).to(torch.int64)
 
     x_view = x.unsqueeze(1)
-    x_floor_view = x_floor.unsqueeze(1)
-    x_ceil_view = x_ceil.unsqueeze(1)
-
-    v1 = input[:, :, x_floor_view, y_floor]
-    v2 = input[:, :, x_ceil_view, y_floor]
-    v3 = input[:, :, x_floor_view, y_ceil]
-    v4 = input[:, :, x_ceil_view, y_ceil]
+    x_floor_view = x_floor.view(1, -1, 1)
+    x_ceil_view = x_ceil.view(1, -1, 1)
+    y_floor_view = y_floor.view(1, 1, -1)
+    y_ceil_view = y_ceil.view(1, 1, -1)
+
+    x_indices = torch.cat([x_floor_view, x_ceil_view, x_floor_view, x_ceil_view], dim=0)
+    y_indices = torch.cat([y_floor_view, y_floor_view, y_ceil_view, y_ceil_view], dim=0)
+    v_indices = torch.arange(4, dtype=torch.int64, device=input.device).view(4, 1, 1)
```

Comment:
This was a tad difficult to read. A better way to write this one would be:
```suggestion
    x_floor_view = x_floor.unsqueeze(1)
    x_ceil_view = x_ceil.unsqueeze(1)
    y_floor_view = y_floor.unsqueeze(0)
    y_ceil_view = y_ceil.unsqueeze(0)

    x_indices = torch.stack([x_floor_view, x_ceil_view, x_floor_view, x_ceil_view], dim=0)
    y_indices = torch.stack([y_floor_view, y_floor_view, y_ceil_view, y_ceil_view], dim=0)
    v_indices = torch.arange(4, dtype=torch.int64, device=input.device)
    v_indices = _unsqueeze_to_dim(v_indices, 3)
```